### PR TITLE
Remove the empty confusing Account section from setting page.

### PR DESCRIPTION
**Pull request by zinsserzh**

### CODE REVIEW

File: js/forum/dist/extension.js [hunk 1]
```diff
@@ -62,6 +63,27 @@ System.register("wuethrich44/sso/main", ["flarum/extend", "flarum/app", "flarum/
                     items.remove('changeEmail');
                     items.remove('changePassword');
                 }
+
+                function checkRemoveAccountSection(items) {
+                    if (items.has('account')) {
```
Comment (fabwu) [inline js/forum/dist/extension.js]: This looks a little bit too complicated to me. Either remove the complete account section or check if there are any children and remove them like this:
```javascript
if (items.get('account').props.children.length === 0) {
    items.remove('account')
}
```

File: js/forum/dist/extension.js [hunk 2]
```diff
@@ -62,6 +63,27 @@ System.register("wuethrich44/sso/main", ["flarum/extend", "flarum/app", "flarum/
                     items.remove('changeEmail');
```
Comment (fabwu) [inline js/forum/dist/extension.js:63]: If you remove the whole account section you can get rid of these function but if future releases add more options to the account section you're not safe 😕

**Review by fabwu — CHANGES_REQUESTED** [review]

File: js/forum/dist/extension.js (hunk 2 above)
Comment (zinsserzh) [reply js/forum/dist/extension.js:63]: No, I'm not simply removing the entire section. The new function is triggered after your `removeProfileActions` function. I check if it is empty. If so, remove, if not, do nothing.

File: js/forum/dist/extension.js (hunk 1 above)
Comment (zinsserzh) [reply js/forum/dist/extension.js]: I submitted a PR https://github.com/flarum/core/pull/1218 to add the `isEmpty()` utility to ItemList. I added the part to check if it is implemented to make sure it is compatible with the current flarum beta7 release. After the PR is added into flarum's core release, the emulation part can be removed. Plus, `ItemList` does not have `props` attribute. My current implementation is the best way I know to [check if an object is empty]( https://stackoverflow.com/questions/679915/how-do-i-test-for-an-empty-javascript-object)

Comment (fabwu) [reply js/forum/dist/extension.js]: I've tested the code above and it works fine in my case. I just don't want to add 20 lines of code only to remove a simple heading. Do you like to simplify this or should we wait until https://github.com/flarum/core/pull/1218 is merged?

**Comment by fabwu** [comment]
@zinsserzh Thanks for this PR. As soon as your flarum core PR is ready I will change the code.

**Comment by zinsserzh** [comment]
Actually this does not rely on the core PR. My previous implementation is wrong. This code is good to go.

**Pull request merged**